backend/lambda_layers/boto_utils/python/boto_utils.py

```diff
@@ -213,12 +213,26 @@ def parse_s3_url(s3_url):
 
 def get_user_info(event):
     req = event.get("requestContext", {})
-    auth = req.get("authorizer", {})
-    claims = auth.get("claims", {})
-    return {
-        "Username": claims.get("cognito:username", "N/A"),
-        "Sub": claims.get("sub", "N/A"),
-    }
+    # If Cognito Auth is being used
+    if "authorizer" in req:
+        auth = req.get("authorizer", {})
+        claims = auth.get("claims", {})
+        return {
+            "Username": claims.get("cognito:username", "N/A"),
+            "Sub": claims.get("sub", "N/A"),
+        }
+    # If Cognito IAM is being used
+    elif "identity" in req:
+        iden = req.get("identity", {})
+        return {
+            "Username": iden.get("userArn", "N/A"),
+            "Sub": iden.get("user", "N/A"),
+        }
+    else:
```

Comment:
Is this else-case an expected situation? (I'm thinking whether we should we be raising an exception/error rather than continuing with missing authn data?)

Reply:
agreed - we should error here. I don't think there's any situation where we'd want to be able to perform operations without knowing whodunit

Reply:
Fair point. I'll make that change (channelling my inner MJ)

Reply:
Just committed a change to add an expection.

Reply:
I have reverted back to inckuding the N/A as the `else` block since we have a unit test already in place (before my changes) that sent an empty request request and expects N/A values back. So reverting back to this to keep the existing behaviour consistent